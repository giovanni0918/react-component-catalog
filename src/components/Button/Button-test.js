import React, { Component } from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('should always render a button', () => {
    const buttons = lockScreen().find("button");
    expect(buttons.length).toBeGreaterThan(0);
  });
});