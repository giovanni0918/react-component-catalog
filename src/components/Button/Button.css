.Button {
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
    min-width: 5rem;
    margin: 0;
    padding: 0.5rem;
    border: 1px solid black;
    border-radius: 2px;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: 500;
    background-color: ghostwhite;
    color: black;
    text-transform: capitalize;
}

.Button--default {
    background-color: grey;
    color: black;
}

.Button--primary {
    background-color: dodgerblue;
    color: black;
}

.Button--info {
    background-color: powderblue;
    color: black;
}

.Button--success {
    background-color: springgreen;
    color: black;
}

.Button--warning {
    background-color: lightgoldenrodyellow;
    color: black;
}

.Button--danger {
    background-color: palevioletred;
    color: black;
}