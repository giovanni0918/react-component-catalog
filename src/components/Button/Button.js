import React, { Component } from 'react';


const Button = ({
  className,
  onClick,
  textContent
}) => (
    <button
      className={className}
      onClick={onClick}>
      {textContent}
    </button>
  );

export default Button;