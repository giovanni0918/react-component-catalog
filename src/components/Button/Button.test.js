import React, { Component } from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('It should always render a button', () => {
  const component = renderer.create(
    <Button
      className='Button Button--primary'
      onClick={() => { console.log('hello') }}
      textContent='hello' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});