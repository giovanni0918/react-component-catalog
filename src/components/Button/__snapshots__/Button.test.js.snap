// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should always render a button 1`] = `
<button
  className="Button Button--primary"
  onClick={[Function]}
>
  hello
</button>
`;
